external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSink.scala

```diff
@@ -30,14 +30,19 @@ private[kafka010] class KafkaSink(
   @volatile private var latestBatchId = -1L
 
   override def toString(): String = "KafkaSink"
+  private val kafkaParams = new ju.HashMap[String, Object](executorKafkaParams)
 
   override def addBatch(batchId: Long, data: DataFrame): Unit = {
     if (batchId <= latestBatchId) {
       logInfo(s"Skipping already committed batch $batchId")
     } else {
       KafkaWriter.write(sqlContext.sparkSession,
-        data.queryExecution, executorKafkaParams, topic)
+        data.queryExecution, kafkaParams, topic)
       latestBatchId = batchId
     }
   }
+
+  override def stop(): Unit = {
+    CachedKafkaProducer.close(kafkaParams)
```

Comment:
This is only closing the producer on the driver, right?  Do we even create on there?